lib/matplotlib/axes/_base.py

```diff
@@ -2379,7 +2369,7 @@ def draw(self, renderer=None, inframe=False):
                 self.patch.get_path(),
                 self.patch.get_transform()))
 
-            renderer.draw_image(gc, round(l), round(b), im)
+            renderer.draw_image(gc, round(l), round(b), data)
```

Comment:
Alternatively, I was thinking that we should only composite images at the *same* zorder.  In the normal case (where the zorder didn't change from the default) that will be the case.  In the case of tinkering with zorder, the user will really care about getting it right and we can either (1) create multiple composites for each zorder "grouping" or (2) just turn off compositing if the images are at different zorders.